components/patient-data/ui/src/main/resources/PhenoTips/PatientConsentConfiguration.xml

```diff
@@ -511,7 +511,24 @@ $xwiki.jsx.use($displayDocName)##
     }
   };
 
+  var __consent_enableRequired = function(trigger, toggleTarget) {
+    if (trigger.selectedIndex &gt;= 0 &amp;&amp; trigger.options[trigger.selectedIndex].value == '1') {
```

Comment:
This should be reversed: if the consent is mandatory it affects all fields, so it doesn't make sense to specify which fields are required.